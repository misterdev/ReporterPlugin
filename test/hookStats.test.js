const { HookStats } = require('../src/HookStats');

describe('HookStats', () => {
  it('should init hooks', () => {
    const hookStats = new HookStats();
    hookStats.initHook('hook1');
    hookStats.initHook('hook2', 2);
    hookStats.initHook('hook3', '2ms');
    expect(hookStats.hooks.hook1).toMatchSnapshot();
    expect(hookStats.hooks.hook2).toMatchSnapshot();
    expect(hookStats.hooks.hook3).toMatchSnapshot();
  });

  it('should increment hook counter', () => {
    const hookStats = new HookStats();
    hookStats.initHook('hook1');
    hookStats.initHook('hook2', 2);
    hookStats.initHook('hook3', '2ms');
    hookStats.incrementCount('hook1');
    hookStats.incrementCount('hook2');
    hookStats.incrementCount('hook3');
    hookStats.incrementCount('hook1');
    hookStats.incrementCount('hook1');
    expect(hookStats.hooks.hook1).toMatchSnapshot();
    expect(hookStats.hooks.hook2).toMatchSnapshot();
    expect(hookStats.hooks.hook3).toMatchSnapshot();
  });

  it('should know when emitting or throttling', () => {
    const hookStats = new HookStats();
    hookStats.initHook('hook1');
    hookStats.initHook('hook2', 2);
    hookStats.initHook('hook3', '2ms');
    expect(`${hookStats.shouldTrigger('hook1')}`).toMatch('true');
    expect(`${hookStats.shouldTrigger('hook2')}`).toMatch('true');
    expect(`${hookStats.shouldTrigger('hook3')}`).toMatch('true');

    hookStats.incrementCount('hook2');
    expect(`${hookStats.shouldTrigger('hook2')}`).toMatch('false');
    hookStats.incrementCount('hook2');
    expect(`${hookStats.shouldTrigger('hook2')}`).toMatch('true');

    hookStats.hooks.hook3.lastCall = Date.now();
    expect(`${hookStats.shouldTrigger('hook3')}`).toMatch('false');
    hookStats.hooks.hook3.lastCall = Date.now() - 2001;
    expect(`${hookStats.shouldTrigger('hook3')}`).toMatch('true');
  });
});
