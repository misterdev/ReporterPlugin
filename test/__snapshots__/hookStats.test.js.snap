// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HookStats should increment hook counter 1`] = `
Object {
  "count": 3,
  "lastCall": 0,
  "name": "hook1",
  "throttle": undefined,
}
`;

exports[`HookStats should increment hook counter 2`] = `
Object {
  "count": 1,
  "lastCall": 0,
  "name": "hook2",
  "throttle": 2,
}
`;

exports[`HookStats should increment hook counter 3`] = `
Object {
  "count": 1,
  "lastCall": 0,
  "name": "hook3",
  "throttle": "2ms",
}
`;

exports[`HookStats should init hooks 1`] = `
Object {
  "count": 0,
  "lastCall": 0,
  "name": "hook1",
  "throttle": undefined,
}
`;

exports[`HookStats should init hooks 2`] = `
Object {
  "count": 0,
  "lastCall": 0,
  "name": "hook2",
  "throttle": 2,
}
`;

exports[`HookStats should init hooks 3`] = `
Object {
  "count": 0,
  "lastCall": 0,
  "name": "hook3",
  "throttle": "2ms",
}
`;
